src/redux/epics/txDialog/submitTransaction.epic.ts

```diff
@@ -70,14 +60,12 @@ export const submitTransactionEpic = (
                     const maxSale = new Amount(amount.muln(1 + buffer)).asString(DECIMALS);
                     tx = api.tx.cennzx.buyAsset(null, fromAsset, toAsset, +sellAmount, 1000000);
                 }
-                const pair = keyring.getPair(signingAccount);
-                if (pair.isLocked) {
-                    pair.decodePkcs8(password);
-                }
-                return tx.signAndSend(pair).pipe(
+                const signer = injector.signer;
+
+                return tx.signAndSend(signingAccount, {signer}).pipe(
                     switchMap(({events, status}: SubmittableResult) => {
-                        if (status.isBroadcast) {
-                            return of(updateTxHash(tx.hash.toString()), updateStage(Stages.Broadcasted));
+                        if (status.isInBlock) {
+                            return of(updateTxHash(status.asInBlock.toString()), updateStage(Stages.Broadcasted));
```

Comment:
```suggestion
                            return of(updateTxHash(status.asInBlock.toString()), updateStage(Stages.isInBlock));
```
or whatever the in block stage is^